dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/resources_plugin/oss.py

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""DolphinScheduler oss resource plugin."""
+from typing import Optional
+from urllib.parse import urljoin
+
+import oss2
+
+from pydolphinscheduler.core.resource_plugin import ResourcePlugin
+from pydolphinscheduler.resources_plugin.base.bucket import Bucket, OSSFileInfo
+
+
+class OSS(ResourcePlugin, Bucket):
+    """OSS object, declare OSS resource plugin for task and workflow to dolphinscheduler.
+
+    :param prefix: A string representing the prefix of OSS.
+    :param access_key_id: A string representing the ID of AccessKey for AliCloud OSS
+
+    to access private files.
+
+    :param access_key_secret: A string representing the secret of AccessKey for AliCloud OSS
+
+    to access private files.
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        access_key_id: Optional[str] = None,
+        access_key_secret: Optional[str] = None,
+        *args,
+        **kwargs
+    ):
+        super().__init__(prefix, *args, **kwargs)
+        self.access_key_id = access_key_id
+        self.access_key_secret = access_key_secret
+
+    _bucket_file_info: Optional[OSSFileInfo] = None
+
+    def get_bucket_file_info(self, path: str):
+        """Get file information from the file url, like repository name, user, branch, and file path."""
+        elements = path.split("/")
+        self.get_index(path, "/", 3)
+        self._bucket_file_info = OSSFileInfo(
+            endpoint="https://" + elements[2].split(".")[1] + ".aliyuncs.com",
+            bucket=elements[2].split(".")[0],
+            file_path="/".join(str(elements[i]) for i in range(3, len(elements))),
+        )
```

Comment:
same here, we should not use too many bare constant here

Reply:
Sorry, I can't think of a better way to parse the url. I want to know your suggestion.

Reply:
I mean we should use the constant variable to replace `/` and `.`, BTW, you should use

```diff
- endpoint="https://" + elements[2].split(".")[1] + ".aliyuncs.com"
- endpoint=f"https://{elements[2].split('.')[1]}.aliyuncs.com"
```
for more pythonic

Reply:
> ```diff
> f"https://{elements[2].split('.')[1]}.aliyuncs.com"
> ```

thanks